tests/tool_use/test_tool_calls.py

```diff
@@ -46,7 +46,8 @@ async def test_tool_call_and_choice(client: openai.AsyncOpenAI):
     assert isinstance(parsed_arguments.get("city"), str)
     assert isinstance(parsed_arguments.get("state"), str)
     assert parsed_arguments.get("city") == "Dallas"
-    assert parsed_arguments.get("state") == "TX"
+    assert parsed_arguments.get("state") == "TX" or parsed_arguments.get(
+        "state") == "Texas"
```

Comment:
This shouldn't be necessary - the schema for the `get_current_weather` tool specifies the following for the state argument:
```
{
"type": "string",
"description": "the two-letter abbreviation for the state "
}
```

If internLM isn't following this, even at `temperature=0` which should be the default for all tool call tests, I'd encourage you to double-check your chat template. Many times, issues with tool call formatting & instruction following boil down to chat template issues.

Reply:
But, given that the schema for the tool call is quite clear, we should avoid loosening the checks for all models because one model struggles to generate it correctly.

Reply:
thanks for your information, after change the description of state from `the two-letter abbreviation for the state` to `must the two-letter abbreviation for the state`, it's works.